test-prow-e2e.sh

```diff
@@ -0,0 +1,149 @@
+        oc logs $pod -n ${NS} > ${ARTIFACT_DIR}/${pod}.logs
+  done
+  oc get serviceaccounts -n ${NS} -o wide > ${ARTIFACT_DIR}/serviceaccount.yaml
+  oc get serviceaccounts -n ${NS} -o yaml >> ${ARTIFACT_DIR}/serviceaccount.yaml
+  oc get console.v1.operator.openshift.io cluster -o yaml >> ${ARTIFACT_DIR}/cluster.yaml
+  
+  if [ -d "$ARTIFACT_DIR" ] && [ -d "$SCREENSHOTS_DIR" ]; then
+    if [[ -z "$(ls -A -- "$SCREENSHOTS_DIR")" ]]; then
+      echo "No artifacts were copied."
+    else
+      echo "Copying artifacts from $(pwd)..."
+      cp -r "$SCREENSHOTS_DIR" "${ARTIFACT_DIR}/screenshots"
+    fi
+  fi
+}
+
+
+
+trap generateLogsAndCopyArtifacts EXIT
+trap generateLogsAndCopyArtifacts ERR
+
+PULL_SECRET_PATH="/var/run/operator-secret/dockerconfig" 
+NAMESPACE="openshift-marketplace"
+SECRET_NAME="ocs-secret"
+NS="openshift-nmstate"
+ARTIFACT_DIR=${ARTIFACT_DIR:=/tmp/artifacts}
+SCREENSHOTS_DIR="cypress/screenshots"
+
+function createSecret {
+    oc create secret generic ${SECRET_NAME} --from-file=.dockerconfigjson=${PULL_SECRET_PATH} --type=kubernetes.io/dockerconfigjson -n $1
+}
+
+function linkSecrets {
+  for serviceAccount in `oc get serviceaccounts -n ${NS} --no-headers -o custom-columns=":metadata.name" | sed 's/"//g'`; do
+        echo "Linking ${SECRET_NAME} to ${serviceAccount}"
+        oc secrets link ${serviceAccount} ${SECRET_NAME} -n ${NS} --for=pull
+  done
+}
+
+function deleteAllPods {
+  oc delete pods --all -n $1 
+}
+
+echo "Creating secret for CI builds in ${NAMESPACE}"
+
+createSecret ${NAMESPACE}
+
+echo "Creating secret for linking pods"
+createSecret ${NS}
+
+echo "Adding secret to all service accounts in ${NS} namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods ${NS}
+
+sleep 30
+
+echo "Adding secret to all service accounts in ${NS} namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods ${NS}
+
+sleep 120
+
+echo "Adding secret to all service accounts in ${NS} namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods ${NS}
+
+sleep 120
+
+# Enable console plugin for nmstate
+export CONSOLE_CONFIG_NAME="cluster"
+export NMSTATE_PLUGIN_NAME="nmstate-console-plugin"
+NMSTATE_PLUGIN_IMAGE="$1"
+
+
+echo ""
+echo "Installing kubernetes-nmstate operator"
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/nmstate.io_nmstates.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/namespace.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/service_account.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/role.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/role_binding.yaml
+kubectl apply -f https://github.com/nmstate/kubernetes-nmstate/releases/download/v0.74.0/operator.yaml
+
+echo "Deploy nmstate console plugin"
```

Comment:
if you install via the operator, you don't install using a manifest, you will need to tell the operator to use the image created using prow